Tools/BiblioTech/CodexCidChecker.cs

```diff
@@ -0,0 +1,198 @@
+
+        private CheckResponse SuccessMessage(DataItem content)
+        {
+            return FormatResponse(
+                success: true,
+                title: $"Success: '{content.Cid}'",
+                error: "",
+                $"size: {content.Manifest.OriginalBytes} bytes",
+                $"blockSize: {content.Manifest.BlockSize} bytes",
+                $"protected: {content.Manifest.Protected}"
+            );
+        }
+
+        private CheckResponse UnexpectedException(Exception ex)
+        {
+            return FormatResponse(
+                success: false,
+                title: "Unexpected error",
+                error: ex.ToString(),
+                content: "Details will be sent to the bot-admin channel."
+            );
+        }
+
+        private CheckResponse UnexpectedReturnCode(string response)
+        {
+            var msg = "Unexpected return code. Response: " + response;
+            return FormatResponse(
+                success: false,
+                title: "Unexpected return code",
+                error: msg,
+                content: msg
+            );
+        }
+
+        private CheckResponse FailedToFetch(string response)
+        {
+            var msg = "Failed to download content. Response: " + response;
+            return FormatResponse(
+                success: false,
+                title: "Could not download content",
+                error: msg,
+                msg,
+                $"Connection trouble? See 'https://docs.codex.storage/learn/troubleshoot'"
+            );
+        }
+
+        private CheckResponse CidFormatInvalid(string response)
+        {
+            return FormatResponse(
+                success: false,
+                title: "Invalid format",
+                error: "",
+                content: "Provided CID is not formatted correctly."
+            );
+        }
+
+        private CheckResponse FormatResponse(bool success, string title, string error, params string[] content)
+        {
+            var msg = string.Join(nl,
+                new string[]
+                {
+                    title,
+                    "```"
+                }
+                .Concat(content)
+                .Concat(new string[]
+                {
+                    "```"
+                })
+            ) + nl + nl;
+
+            return new CheckResponse(success, msg, error);
+        }
+
+        #endregion
+
+        #region Codex Node API 
+
+        private CodexApi GetCodex()
+        {
+            if (currentCodexNode == null) currentCodexNode = CreateCodex();
+            return currentCodexNode;
+        }
+
+        private async Task<bool> CheckCodex(CodexApi codex)
+        {
+            try
+            {
+                var info = await currentCodexNode!.GetDebugInfoAsync();
+                if (info == null || string.IsNullOrEmpty(info.Id)) return false;
+                return true;
+            }
+            catch (Exception e)
```

Comment:
This can easily draw you into bugs that are hard to figure out if that inner expression is raising an exception for a reason that's different from the one you'd expect. I'd use a narrower exception that actually signals failure to connect to Codex, if that exists (as you did above).